internal/system/agent/executor/docker_exec.go

```diff
@@ -0,0 +1,134 @@
+package executor
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/edgexfoundry/edgex-go/internal"
+	"github.com/edgexfoundry/edgex-go/internal/system/agent/logger"
+)
+
+var services = map[string]string{
+	internal.SupportNotificationsServiceKey: "edgex-support-notifications",
+	internal.CoreDataServiceKey:             "edgex-core-data",
+	internal.CoreMetaDataServiceKey:         "edgex-core-metadata",
+	internal.CoreCommandServiceKey:          "edgex-core-command",
+	internal.ExportClientServiceKey:         "edgex-export-client",
+	internal.ExportDistroServiceKey:         "edgex-export-distro",
+	internal.SupportLoggingServiceKey:       "edgex-support-logging",
+	internal.ConfigSeedServiceKey:           "edgex-config-seed",
+}
+
+func FindDockerContainerStatus(service string, status string) bool {
+
+	var (
+		cmdOut []byte
+		err    error
+	)
+	cmdName := "docker"
+	cmdArgs := []string{"ps"}
+	if cmdOut, err = exec.Command(cmdName, cmdArgs...).Output(); err != nil {
+		logs.LoggingClient.Error("error running the docker command", "error message", err.Error())
+		os.Exit(1)
+	}
+	dockerOutput := string(cmdOut)
+	// Find whether the container to start has started.
+	for _, line := range strings.Split(strings.TrimSuffix(dockerOutput, "\n"), "\n") {
+		if strings.Contains(line, service) {
+
+			if status == "Up" {
+				if strings.Contains(line, "Up") {
+					logs.LoggingClient.Debug("container started", "service name", service, "details", line)
+					return true
+				} else {
+					logs.LoggingClient.Warn("container NOT started", "service name", service)
+					return false
+				}
+			} else if status == "Exited" {
+				if strings.Contains(line, "Exited") {
+					logs.LoggingClient.Debug("container stopped", "service name", service, "details", line)
+					return true
+				} else {
+					logs.LoggingClient.Warn("container NOT stopped", "service name", service)
+					return false
+				}
+			}
+		}
+	}
+	return false
+}
+
+func ExecuteDockerCommands(service string, operation string) error {
+	_, knownService := services[service]
+
+	if knownService {
+		RunDockerCommands(service, services[service], operation)
+
+		return nil
+	} else {
+		newError := fmt.Errorf("unknown service: %v", service)
+		logs.LoggingClient.Error(newError.Error())
+
+		return newError
+	}
+}
+
+func RunDockerCommands(service string, dockerService string, operation string) {
```

Comment:
Function isn't called outside of this package, so doesn't need to be exported (capitalized)